File: 0605-can-place-flowers/0605-can-place-flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function(flowerbed, n) {
    for(let i = 0; i < flowerbed.length; i++){
        if(!flowerbed[i - 1] && !flowerbed[i] && !flowerbed[i + 1]){
            flowerbed[i] = 1;
            n -= 1;
        }
    };

    return n < 1
};

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        m = len(flowerbed)
        flowerbed.append(0)
        curr = 0

        for i in range(m):
            prev = curr
            curr = flowerbed[i]

            if curr:
                continue

            if not prev and not flowerbed[i+1]:
                n -= 1
                curr = 1
        
        return n <= 0
            